rpcs3/Emu/Cell/SPUThread.cpp

```diff
@@ -1550,11 +1627,88 @@ bool SPUThread::set_ch_value(u32 ch, u32 value)
 		// Reset stall status for specified tag
 		if (atomic_storage<u32>::btr(ch_stall_mask.raw(), value))
 		{
-			auto mfc = fxm::check_unlocked<mfc_thread>();
+			u32 removed = 0;
+			u32 barrier = ch_stall_mask;
+
+			// Process enqueued commands
+			std::remove_if(mfc_queue + 0, mfc_queue + mfc_size, [&](spu_mfc_cmd& args)
+			{
+				if ((args.cmd & ~0xc) == MFC_BARRIER_CMD)
+				{
+					if (&args - mfc_queue <= removed)
+					{
+						barrier = ch_stall_mask;
+						_mm_mfence();
+						removed++;
+						return true;
+					}
+
+					barrier |= 0xffffffffu;
+					return false;
+				}
+
+				if ((1u << args.tag) & barrier)
+				{
+					return false;
+				}
+
+				if (args.cmd & MFC_LIST_MASK)
+				{
+					if (do_list_transfer(args))
+					{
+						removed++;
+						return true;
+					}
 
-			//if (test(mfc->state, cpu_flag::is_waiting))
+					barrier |= (1u << args.tag);
+					return false;
+				}
+
+				if ((args.cmd & ~(MFC_BARRIER_MASK | MFC_FENCE_MASK)) == MFC_PUTQLLUC_CMD)
```

Comment:
you can drop the `& ~(MFC_BARRIER_MASK | MFC_FENCE_MASK))` as theres no a barrier or a fence version of the putqlluc command